### PR TITLE
Allowlist

### LINKED ISSUE #547: Invite-only registration
Admin can invite users via email using a separate endpoint and `x-hasura-admin-secret`.

### PR DESCRIPTION
**Pull request by komninoschat**
Fixes #547.

### CODE REVIEW

File: src/shared/config/authentication/registration.ts [hunk 1]
```diff
@@ -31,6 +31,9 @@ export const REGISTRATION = {
   get ALLOWED_USER_ROLES() {
     return castStringArrayEnv('ALLOWED_USER_ROLES', this.DEFAULT_ALLOWED_USER_ROLES)
   },
+  get ALLOWLIST() {
+    return castBooleanEnv('ALLOWLIST_ENABLE', false)
```
Comment (elitan) [inline src/shared/config/authentication/registration.ts:35]: `_ENABLED`

**Review by elitan — CHANGES_REQUESTED** [review]
- invite email template
- env var if invite email template should be sent or not. Default to `true`.

**Comment by elitan** [comment]
Do we want to change the table name and endpoint from `allowlist` to something more related to **registration** specifically?

Naming it `auth.allowlist` does not make it clear that it's related to allow registrations. Some people might belive it's related to allow people to login, or to allow people to change password or something else?

**Comment by nikolowry** [comment]
Sorry for commenting again on an already merged PR, but it be great if either @komninoschat  or @elitan could provide some clarifications on the following:
-  Is the final namespace of this feature settled on? I'm just a little confused with this PR being named "Allowlist" but all the ENV vars being prefixed with `WHITELIST`. Also, https://github.com/nhost/hasura-backend-plus/pull/555/files was closed without being merged or commented on. Maybe I missed the discussion, but is there a link to where this decision was discussed and reached? 
- Semi-related, but how long will `_ENABLED` and `_ENABLE` both be supported? There's a lack of consistency in the docs about which one we should ultimately prefer.

I really appreciate the project and how quick you've guys have been with communication and fixes. Thanks for the hard work!

**Comment by elitan** [comment]
- It's going to be named [whitelist](https://en.wikipedia.org/wiki/Whitelisting). 
- We're moving from `_ENABLE` to `_ENABLED` so it's safe to start using `_ENABLED`.

Thanks for your kind words.

**Pull request merged**